src/Orleans.Transactions/DistributedTM/TransactionAgent.cs

```diff
@@ -105,14 +105,17 @@ private async Task<TransactionalStatus> CommitReadOnlyTransaction(TransactionInf
             }
 
             // examine the return status
-            if (status == TransactionalStatus.Ok) foreach (var s in tasks)
+            if (status == TransactionalStatus.Ok)
             {
-                status = s.Result;
-                if (status != TransactionalStatus.Ok)
+                foreach (var s in tasks)
                 {
-                    if (logger.IsEnabled(LogLevel.Debug))
-                        logger.Debug($"{stopwatch.Elapsed.TotalMilliseconds:f2} fail {transactionInfo.TransactionId} prepare response status={status}");
-                    break;
+                    status = s.Result;
```

Comment:
In general we should use `.GetAwaiter().GetResult()` because `.Result` unconditionally wraps any exceptions from the task in an `AggregateException`